pulsar/connector/consumer.go

```diff
@@ -37,19 +92,31 @@ func WithMessageChannel(messageChannel chan pulsar.ConsumerMessage) CreateConsum
 	}
 }
 
-func CreateConsumer(pulsarClient pulsar.Client, createConsumerOpts ...CreateConsumerOption) (pulsar.Consumer, error) {
+func CreateSharedConsumer(pulsarClient pulsar.Client, createConsumerOpts ...CreateConsumerOption) (pulsar.Consumer, error) {
 	opts := &createConsumerOptions{}
+	opts.defaults()
 	for _, o := range createConsumerOpts {
 		o(opts)
 	}
+	if err := opts.validate(); err != nil {
+		return nil, err
+	}
+	var dlq *pulsar.DLQPolicy
+	if opts.MaxDeliveryAttempts != 0 {
+		dlq = NewDlq(opts.Topic, opts.MaxDeliveryAttempts)
+	}
+	return pulsarClient.Subscribe(pulsar.ConsumerOptions{
+		Topic:                          GetTopic(opts.Topic),
```

Comment:
why this is a global function? this should be a method of the client.

Reply:
It can be used if someone want to create a producer outside the client